menu.kando.Kando.yml

```diff
@@ -0,0 +1,35 @@
+id: menu.kando.Kando
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+command: AppRun
+finish-args:
+  - --share=ipc
+  - --device=dri
+  - --socket=x11
+  - --socket=wayland
+  - --talk-name=org.freedesktop.Flatpak
+  - --talk-name=org.gnome.Shell
+  - --talk-name=org.kde.KWin
+modules:
+  - name: kando
+    buildsystem: simple
+    sources:
+      - type: file
+        dest-filename: kando.AppImage
+        url: https://github.com/kando-menu/kando/releases/download/v1.6.0/Kando-1.6.0-x86_64.AppImage
```

Comment:
why not build from source?

Reply:
As mentioned above, I have tried this. But Kando is not a simple Electron project: It ships several C++ backends for various Wayland compositors which are built using [CMake.js](https://github.com/cmake-js/cmake-js). During the build process, CMake.js tries to download stuff from the internet which is not allowed on Flathub. Hence I am forced to package the prebuilt binary from the corresponding GitHub release.

Reply:
maybe that where you fix the tool to allow an offline mode. All that is needed is a list of what it downloads and then know where to put it so it can find it. Even npm and yarn allow that. So I don't get each time the wheel is reivented it has to address all the same problem.

Reply:
I tried this now for a couple of hours and - well, CMake.js does not seem to be the only issue here. It seems that `flatpak-node-generator` does [not properly support package-lock files with version 3](https://github.com/flatpak/flatpak-builder-tools/issues/366). I also encounter [this issue](https://github.com/flatpak/flatpak-builder-tools/issues/388).

If I downgrade to package-lock version 2, I get issues with an package which is downloaded from a git source. Maybe [this issue](https://github.com/flatpak/flatpak-builder-tools/issues/381)? There also seems to be a [corresponding merge request](https://github.com/flatpak/flatpak-builder-tools/pull/382) which has not been merged yet.

So I would really appreciate if we could simply bundle the binary build on GitHub...

Reply:
or help fixing these issues. apparently everyone complain, but no one to fix them.

Reply:
I would love to. But I have several other opens source projects to maintain and only very little spare time (full-time job, family, you know). So I sadly do not have the time to fix tools which seem to be required for publishing Electron apps on Flathub (and this would take _weeks_ given the knowledge I have regarding said tools and the daily time budget I have).